docs/development/Defines.md

```diff
@@ -0,0 +1,533 @@
+# Defines
+
+Betaflight uses `#DEFINE` syntax to include software components to support additional features at build time.
+
+A basic set of defines is included in the config file for the board. At build time, the user can add extra defines in the the Configurator's Firmware Flasher Tab.
+
+When selecting GPS, Baro, etc from the `Other Options` list the required additional defines are added automatically. For example, if `BARO` is requested by the user, the `USE_BARO` define, and all supported baro hardware defines are automatically included.
+
+Additional defines may be added:
+
+- to a cloud build, by entering the name of the define, without the `USE_` prefix, separated by spaces, in the `Custom Defines` list, e.g. `ANA EXTI`.
+- to a local build, by adding the define in a syntax matching the other defines,
+  e.g. `-DUSE_ANA -DUSE_EXTI`
+
+A full list of defines as of MAY 2024 is provided below. The list was generated using `grep --exclude-dir=config -Irh '^[^/].*' src/ | grep -wo 'USE_[0-9A-Z_]\+' | sort | uniq` - thanks @nerdcopter!
+
+:::note
+
+Not all the defines in this list are 'safe'; some are archaic and some are not actively maintained.
+Defines on commented-out lines are not included in the list.
+Some defines are for internal register definitions and are not end-user code options.
```

Comment:
@ctzsnooze suggest leaving out middleware definitions.

Reply:
By prepping only src/main we remove all middleware `USE_` defines.  The only unnecessary ones are those where an internal register definition commences with `USE_`.  Fortunately, not many of them.
Updated the listing now.